app/models/style.rb

```diff
@@ -1,10 +1,19 @@
 class Style < ActiveRecord::Base
   self.table_name = 'style'
   validates_presence_of :path
+  validate :path_is_css
+
   has_many :displays, foreign_key: :style
 
   def descr
     # Dumb thing has HTML in it, but it's safe
     read_attribute(:descr).html_safe
   end
+
+  private
+  def path_is_css
+      if path && !path.end_with?('.css')
```

Comment:
Since we are validating that `path` is always present (and I believe that validation will be run first), you can simplify this a little by removing that first condition.

Reply:
👍 I suspected that the validation would be run first and this wouldn't be necessary, but good to have that semi-confirmed. Fixed.

Reply:
Haha, according to our failing specs it appears I have misled you. I'm not sure why that doesn't work the way I expect, but I'll just merge in the previous version with the extra check, since it won't do any harm.